src/client/app/effects/snippet.ts

```diff
@@ -372,14 +360,41 @@ export class SnippetEffects {
         properties['mode'] = isViewMode ? 'view' : 'editor';
         AI.trackEvent(mode, properties);
 
-        /*
-         * If a imported snippet is a SAMPLE or the app is in view mode, then skip the save (simply to avoid clutter).
-         * The snippet will get saved as soon as the user makes any changes (if in editor mode).
+        /**
+         * If the action here involves true importing rather than re-opening,
+         * and if the name is already taken by a local snippet, ask the user if they want to:
+         * import the snippet with a new name, go to the already existing snippet, or cancel the import
          */
-        if (mode !== Snippet.ImportType.SAMPLE && !isViewMode) {
-            actions.push(new Snippet.SaveAction(snippet));
+        let importResult = null;
+        if (mode !== Snippet.ImportType.OPEN && this._exists(snippet.name)) {
+            importResult = await this._uiEffects.alert(
```

Comment:
Could we get this scenario (import existing already-in-workspace gist) into the markdown file that has the list of scenarios to test before deployment?